src/Symfony/Component/HttpClient/Internal/AmpClientState.php

```diff
@@ -203,13 +189,11 @@ private function handlePush(Request $request, Promise $response, array $options)
 
         $url = (string) $request->getUri();
         $this->logger?->debug(sprintf('Queueing pushed response: "%s"', $url));
-        $this->pushedResponses[$authority][] = [$url, $deferred, $request, $response, [
+        $this->pushedResponses[$authority][] = [$url, $response, $request, [
             'proxy' => $options['proxy'],
             'bindto' => $options['bindto'],
             'local_cert' => $options['local_cert'],
             'local_pk' => $options['local_pk'],
         ]];
-
-        return $deferred->promise();
```

Comment:
this maps to my note on AmpHttpClient: how can I cancel a pushed response from outside the push-handler?

Reply:
I need to take a look myself, but the equivalent to the previous version would be replacing `Deferred` with `DeferredFuture` and then this line with `$deferred->getFuture()->await()` and then call `error` on the deferred for rejection.

Reply:
Similar to before. Return the future from a `DeferredFuture` in the push handler, then later either resolve it with null to accept the push or fail it with `CancelledException` to reject the push.

@kelunik What do you think of passing a `DeferredCancellation` to the push handler to provide a more direct way to cancel the push? We already have an instance, so the question is do we want to expose it or not?

Reply:
Thanks, I think I now get it. PR updated, see [3rd commit](https://github.com/nicolas-grekas/symfony/pull/43/commits/ebfd618abee97fc25f3d46ae70801c17b62e43d1)
Please let me know if I made a mistake.